phoenix-core/src/main/java/org/apache/phoenix/coprocessor/IndexToolVerificationResult.java

```diff
@@ -413,13 +413,8 @@ public void update(Cell cell) {
     }
 
     public boolean isVerificationFailed(IndexTool.IndexVerifyType verifyType) {
-        if (verifyType == IndexTool.IndexVerifyType.BEFORE || verifyType == IndexTool.IndexVerifyType.NONE) {
+        if (verifyType == IndexTool.IndexVerifyType.BEFORE || verifyType == IndexTool.IndexVerifyType.NONE || verifyType == IndexTool.IndexVerifyType.ONLY) {
             return false;
-        } else if (verifyType == IndexTool.IndexVerifyType.ONLY) {
-            if (getBefore().getInvalidIndexRowCount() + getBefore().getMissingIndexRowCount() +
-                before.getBeyondMaxLookBackInvalidIndexRowCount() + before.getBeyondMaxLookBackMissingIndexRowCount() > 0) {
-                return true;
-            }
         } else if (verifyType == IndexTool.IndexVerifyType.BOTH || verifyType == IndexTool.IndexVerifyType.AFTER) {
```

Comment:
I suggest the following instead of above
public boolean isVerificationFailed() {
  if (getAfter().getInvalidIndexRowCount() + getAfter().getMissingIndexRowCount() +
                after.getBeyondMaxLookBackInvalidIndexRowCount() + after.getBeyondMaxLookBackMissingIndexRowCount() > 0) {
                return true;
            }
  return false;
}

Reply:
nice, however, we ll still need a check on "both" and "after"